### ISSUE TITLE
[BUG] AAS-ENV fails to start with milestone-03 - identical config to milestone-02

### ISSUE THREAD

**Issue by gerberm-at-msg** [eclipse-basyx/basyx-java-server-sdk#378]
**Describe the bug**
I updated the image in the docker compose from milestone-02 to milestone-03
image: eclipsebasyx/aas-environment:2.0.0-milestone-03

**To Reproduce**

Docker is configured to an IP-Address in my case 192.168.178.74
```
environment:
      AAS_REGISTRY_PATH: http://192.168.178.75:8082/shell-descriptors
      SUBMODEL_REGISTRY_PATH: http://192.168.178.75:8083/submodel-descriptors
      AAS_REPO_PATH: http://192.168.178.75:8081/shells
      SUBMODEL_REPO_PATH: http://192.168.178.75:8081/submodels
      CD_REPO_PATH: http://192.168.178.75:8081/concept-descriptions
      AAS_DISCOVERY_PATH: http://192.168.178.75:8084/lookup/shells
```

aas-env.propierties config is:
```
server.port=8081

basyx.backend=MongoDB

spring.data.mongodb.host=mongo
spring.data.mongodb.database=aas-env
spring.data.mongodb.authentication-database=admin
spring.data.mongodb.username=mongoAdmin
spring.data.mongodb.password=mongoPassword

mqtt.clientId=AAS-Env-8081
mqtt.hostname=mosquitto
mqtt.port=1883

basyx.aasrepository.feature.mqtt.enabled=true
basyx.submodelrepository.feature.mqtt.enabled=true
basyx.environment=file:aas
basyx.cors.allowed-origins=*
basyx.cors.allowed-methods=GET,POST,PATCH,DELETE,PUT,OPTIONS,HEAD

basyx.aasrepository.feature.registryintegration=http://aas-registry:8080
basyx.submodelrepository.feature.registryintegration=http://sm-registry:8080
basyx.externalurl=http://192.168.178.75:8081
```

Additional the error message in docker is:
```
2024-08-02T09:54:38.398Z  WARN 1 --- [AAS Environment] [           main] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'simpleAasRepositoryFactory' defined in URL [jar:nested:/application/basyxExecutable.jar/!BOOT-INF/lib/basyx.aasrepository-backend-2.0.0-milestone-03.jar!/org/eclipse/digitaltwin/basyx/aasrepository/backend/SimpleAasRepositoryFactory.class]: Unsatisfied dependency expressed through constructor parameter 0: Error creating bean with name 'aasMongoDBBackendProvider': Resolution of declared constructors on bean Class [org.eclipse.digitaltwin.basyx.aasrepository.backend.mongodb.AasMongoDBBackendProvider] from ClassLoader [org.springframework.boot.loader.launch.LaunchedClassLoader@2ff4acd0] failed
Related cause: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'simpleAasRepositoryFactory' defined in URL [jar:nested:/application/basyxExecutable.jar/!BOOT-INF/lib/basyx.aasrepository-backend-2.0.0-milestone-03.jar!/org/eclipse/digitaltwin/basyx/aasrepository/backend/SimpleAasRepositoryFactory.class]: Unsatisfied dependency expressed through constructor parameter 0: Error creating bean with name 'aasMongoDBBackendProvider': Resolution of declared constructors on bean Class [org.eclipse.digitaltwin.basyx.aasrepository.backend.mongodb.AasMongoDBBackendProvider] from ClassLoader [org.springframework.boot.loader.launch.LaunchedClassLoader@2ff4acd0] failed
2024-08-02T09:54:38.401Z  INFO 1 --- [AAS Environment] [           main] o.apache.catalina.core.StandardService   : Stopping service [Tomcat]
2024-08-02T09:54:38.435Z  INFO 1 --- [AAS Environment] [           main] .s.b.a.l.ConditionEvaluationReportLogger : 

Error starting ApplicationContext. To display the condition evaluation report re-run your application with 'debug' enabled.
2024-08-02T09:54:38.454Z ERROR 1 --- [AAS Environment] [           main] o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'simpleAasRepositoryFactory' defined in URL [jar:nested:/application/basyxExecutable.jar/!BOOT-INF/lib/basyx.aasrepository-backend-2.0.0-milestone-03.jar!/org/eclipse/digitaltwin/basyx/aasrepository/backend/SimpleAasRepositoryFactory.class]: Unsatisfied dependency expressed through constructor parameter 0: Error creating bean with name 'aasMongoDBBackendProvider': Resolution of declared constructors on bean Class [org.eclipse.digitaltwin.basyx.aasrepository.backend.mongodb.AasMongoDBBackendProvider] from ClassLoader [org.springframework.boot.loader.launch.LaunchedClassLoader@2ff4acd0] failed
        at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:795) ~[spring-beans-6.1.11.jar!/:6.1.11]
        at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:237) ~[spring-beans-6.1.11.jar!/:6.1.11]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1375) ~[spring-beans-6.1.11.jar!/:6.1.11]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1212) ~[spring-beans-6.1.11.jar!/:6.1.11]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:562) ~[spring-beans-6.1.11.jar!/:6.1.11]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:522) ~[spring-beans-6.1.11.jar!/:6.1.11]
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:337) ~[spring-beans-6.1.11.jar!/:6.1.11]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-6.1.11.jar!/:6.1.11]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:335) ~[spring-beans-6.1.11.jar!/:6.1.11]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200) ~[spring-beans-6.1.11.jar!/:6.1.11]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:975) ~[spring-beans-6.1.11.jar!/:6.1.11]
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:971) ~[spring-context-6.1.11.jar!/:6.1.11]
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:625) ~[spring-context-6.1.11.jar!/:6.1.11]
        at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146) ~[spring-boot-3.3.2.jar!/:3.3.2]
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754) ~[spring-boot-3.3.2.jar!/:3.3.2]
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:456) ~[spring-boot-3.3.2.jar!/:3.3.2]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:335) ~[spring-boot-3.3.2.jar!/:3.3.2]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1363) ~[spring-boot-3.3.2.jar!/:3.3.2]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1352) ~[spring-boot-3.3.2.jar!/:3.3.2]
        at org.eclipse.digitaltwin.basyx.aasenvironment.component.AasEnvironmentComponent.main(AasEnvironmentComponent.java:42) ~[!/:2.0.0-milestone-03]
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[na:na]
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
        at java.base/java.lang.reflect.Method.invoke(Method.java:569) ~[na:na]
        at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:91) ~[basyxExecutable.jar:2.0.0-milestone-03]
        at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:53) ~[basyxExecutable.jar:2.0.0-milestone-03]
        at org.springframework.boot.loader.launch.JarLauncher.main(JarLauncher.java:58) ~[basyxExecutable.jar:2.0.0-milestone-03]
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'aasMongoDBBackendProvider': Resolution of declared constructors on bean Class [org.eclipse.digitaltwin.basyx.aasrepository.backend.mongodb.AasMongoDBBackendProvider] from ClassLoader [org.springframework.boot.loader.launch.LaunchedClassLoader@2ff4acd0] failed
```

**Expected behavior**
AAS Environment docker would start

**Screenshots**

**BaSyx configuration**
Provide details about the configuration you used. If possible, attach the configuration as a .zip file.

**System (please complete the following information):**
 - OS: Linux debian 12
 - CPU architecture: x64
 - Browser ----
 - Version ----

**Additional context**

**Comment by aaronzi** [comment]
Thank you for raising this issue @gerberm-at-msg.
We will have a look and come back to you with a solution.

**Comment by FriedJannik** [comment]
Hi @gerberm-at-msg,
Thank you for the Report.
This was indeed a bug which was fixed in the PR mentioned above #383 
We are planning to release a new image with the fix included.

**Comment by mdanish98** [comment]
Hi @gerberm-at-msg ,

Thanks for your patience, we fixed this issue and released a new image (2.0.0-milestone-03.1) with this fix. 
Here is the latest image: https://hub.docker.com/r/eclipsebasyx/aas-environment/tags

docker pull eclipsebasyx/aas-environment:2.0.0-milestone-03.1

Could you please check this image?

**Comment by gerberm-at-msg** [comment]
Hi @mdanish98, 
it works, thank you!
Will open a new ticket for another issue reg. keycloak soon - has nothing to do with this one. 
Cheers
Michael
